src/main/java/com/github/steveice10/mc/auth/service/MsaAuthenticationService.java

```diff
@@ -38,13 +38,14 @@ public class MsaAuthenticationService extends AuthenticationService {
 
     private String deviceCode;
     private String clientId;
+    private String clientSecret;
     private String refreshToken;
 
-    public MsaAuthenticationService(String clientId) {
-        this(clientId, null);
+    public MsaAuthenticationService(String clientId, String clientSecret) {
```

Comment:
There should be a new constructor and not adding to an existing so we don't break software using the library

Reply:
Pushed new code where the client_secret is optional,

Maded changes on each request that can use a client_secret, the requests will only include a client_secret when this is set.
As stated above you can now use it as:

```java
MsaAuthenticationService service = new MsaAuthenticationService("<clientID>");
            service.setClientSecret("<secret>");
            service.login();
```

PS: it's my first ever contribution so i appricate the feedback!
Let me know if something can be diffrent / better!